tests/scalers/temporal/temporal_test.go

```diff
@@ -0,0 +1,265 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{.TemporalDeploymentName}}
+  namespace: {{.TestNamespace}}
+  labels:
+    app: temporal
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: temporal
+  template:
+    metadata:
+      labels:
+        app: temporal
+    spec:
+      containers:
+        - name: ubuntu
+          image: ubuntu:latest
+          command: ["bash", "-c"]
+          args:
+            - |
+              apt-get update && apt-get install -y curl && \
+              curl -sSf https://temporal.download/cli.sh | sh && \
+              PATH="\$PATH:/root/.temporalio/bin" >> ~/.bashrc && \
+              source ~/.bashrc && \
+              temporal server start-dev --namespace v2 --ip 0.0.0.0
+          ports:
+            - containerPort: 7233
+          livenessProbe:
+            exec:
+              command:
+                - /bin/sh
+                - -ec
+                - test $(ps -ef | grep -v grep | grep "temporal server" | wc -l) -eq 1
+            failureThreshold: 5
+            initialDelaySeconds: 10
+            periodSeconds: 30
+            successThreshold: 1
+            timeoutSeconds: 1
+          readinessProbe:
+            exec:
+              command:
+                - /bin/sh
+                - -ec
+                - test $(ps -ef | grep -v grep | grep "temporal server" | wc -l) -eq 1
+            failureThreshold: 5
+            initialDelaySeconds: 10
+            periodSeconds: 30
+            successThreshold: 1
+            timeoutSeconds: 1
+`
+
+	scaledObjectTemplate = `
+apiVersion: keda.sh/v1alpha1
+kind: ScaledObject
+metadata:
+  name: {{.ScaledObjectName}}
+  namespace: {{.TestNamespace}}
+spec:
+  scaleTargetRef:
+    name: {{.DeploymentName}}
+  pollingInterval: 5
+  cooldownPeriod:  10
+  minReplicaCount: 0
+  maxReplicaCount: 1
+  advanced:
+    horizontalPodAutoscalerConfig:
+      behavior:
+        scaleDown:
+          stabilizationWindowSeconds: 10
+  triggers:
+  - type: temporal
+    metadata:
+      namespace: default
+      queueName: hello-task-queue
+      targetQueueSize: "2"
+      activationTargetQueueSize: "3"
+      endpoint: {{.TemporalDeploymentName}}.{{.TestNamespace}}.svc.cluster.local:7233
+`
+
+	deploymentTemplate = `
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{.DeploymentName}}
+  namespace: {{.TestNamespace}}
+  labels:
+    app: {{.DeploymentName}}
+spec:
+  replicas: 0
+  selector:
+    matchLabels:
+      app: {{.DeploymentName}}
+  template:
+    metadata:
+      labels:
+        app: {{.DeploymentName}}
+    spec:
+      containers:
+      - name: nginx
+        image: nginxinc/nginx-unprivileged
+        ports:
+        - containerPort: 80
+`
+
+	jobWorkerTemplate = `
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: worker
+  namespace: {{.TestNamespace}}
+spec:
+  template:
+    spec:
+      containers:
+      - name: worker
+        image: "prajithp/temporal-sample:1.0.0"
```

Comment:
Please use `https://github.com/temporalio/omes` for workers and to start workflows as required. If you need some support/feature for something there, please let me know and/or file issues.

Reply:
If you could provide a Docker image for this, that would be great. Otherwise, I'll have to go through the codebase. If you're okay with it, we can proceed with the existing setup, and I'll raise a PR for the Dockerfile to https://github.com/kedacore/test-tools.

Reply:
Images are available here: https://hub.docker.com/r/temporaliotest/omes

Reply:
I can give you some commands to run a worker/starter, will get back to you with those.

Reply:
updated it to use [temporaliotest/omes](https://hub.docker.com/r/temporaliotest/omes) @robholland

Reply:
@robholland can you please review this?